src/clj/puppetlabs/services/jruby/jruby_puppet_core.clj

```diff
@@ -198,13 +198,25 @@
 
 (schema/defn initialize-and-create-jruby-config :- jruby-schemas/JRubyConfig
   "Handles the initialization of the jruby-puppet config for the purpose of returning a
-  jruby config"
-  ([raw-config :- {schema/Keyword schema/Any}]
-   (initialize-and-create-jruby-config raw-config nil (fn [])))
-  ([raw-config :- {schema/Keyword schema/Any}
-    profiler :- (schema/maybe PuppetProfiler)
-    agent-shutdown-fn :- IFn]
-   (initialize-and-create-jruby-config raw-config profiler agent-shutdown-fn true))
+  jruby config. This function will only use the :jruby-puppet and :http-client sections
+  from raw-config. If values are not provided, everything in :http-client :jruby-puppet
+  will be given default values, except for :ruby-load-path and :gem-home in the
+  :jruby-puppet subsection.
+
+  The 1-arity function takes only a config and supplies a default of nil for the profiler,
+  an empty fn for the agent-shutdown-fn, and suppresses warnings about legacy auth.conf.
+  This arity is indended for uses where a jruby-config is required but will not be used
+  to to create a pool, such as the cli ruby subcommands
+
+  The 4-arity function takes a profiler object which will be placed into the puppetserver
+  config through the :initialize-pool-instance lifecycle function. The agent-shutdown-fn
+  is run when a jruby-instance is terminated. When warn-legacy-auth-conf? is passed in as
+  true, it will log a warning that the use-legacy-auth-conf setting is depreciated if the
+  config setting is set to true as well."
+  ([raw-config :- {:jruby-puppet {schema/Keyword schema/Any}
+                   (schema/optional-key :http-client) {schema/Keyword schema/Any}
```

Comment:
I think there maybe still needs to be a change the schema for `extract-http-config` to allow nil, because otherwise if this key isn't present then nil will be passed to it, which will make it unhappy.